pkgs/by-name/lm/lm_sensors/package.nix

```diff
@@ -2,41 +2,63 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   bash,
   bison,
   flex,
   which,
   perl,
+  rrdtool,
   sensord ? false,
-  rrdtool ? null,
 }:
 
-assert sensord -> rrdtool != null;
+let
+  version = "3.6.0";
+  tag = lib.replaceStrings [ "." ] [ "-" ] version;
+in
 
 stdenv.mkDerivation rec {
   pname = "lm-sensors";
-  version = "3.6.0";
-  dashedVersion = lib.replaceStrings [ "." ] [ "-" ] version;
+  inherit version;
 
   src = fetchFromGitHub {
     owner = "lm-sensors";
     repo = "lm-sensors";
-    rev = "V${dashedVersion}";
+    inherit tag;
     hash = "sha256-9lfHCcODlS7sZMjQhK0yQcCBEoGyZOChx/oM0CU37sY=";
   };
 
+  patches = lib.optionals sensord [
+    # Fix compile failure on GCC 14 with `sensord` enabled.
+    # From: https://github.com/lm-sensors/lm-sensors/pull/483
+    (fetchpatch {
+      url = "https://github.com/lm-sensors/lm-sensors/pull/483/commits/7a6170f07d05cc6601b4668f211e9389f2e75286.patch";
+      hash = "sha256-Q49quv3eXeMvY3jgZFs/F7Rljbq4YyehIDIlsgmloBQ=";
+    })
+  ];
+
+  outputs = [
+    "bin"
```

Comment:
5 outputs are a lot for such a small tool. We can probably drop the bin output without any impact

Reply:
The split of `bin` and `out` is the main purpose of this PR. It removes `perl` dependency for downstream packages that only want libsensors, which is the main use case from [a quick search](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20lm_sensors&type=code).
I personally want to remove "doc" but I want to wait for opinion from @ peterhoeg